CONTRIBUTING.md

```diff
@@ -55,131 +54,265 @@ Please follow these steps and note these guidelines to begin contributing:
 
-   rect {
-        fill:  ${props => props.theme.DarkTheme ? "#313131" : "#C9FCF6"};  
-      }
-      path {
-        stroke: ${props => props.theme.DarkTheme ? props.theme.keppelColor: "#00B39F"};
+[Example of image file with transparent sections](https://github.com/layer5io/layer5/blob/master/src/assets/images/app/projects/meshery-logo-light.png)
+
+You can see these two examples in action [here](https://layer5.io/projects) by changing the color modes.
+
+### Changing SVG according to color mode
+
+SVGs are preferable to image files because we have the ability to control colors used to fill path or stroke using styled-components and have them change dependent on the color mode without using a conditional.
+
+#### Importing SVG as a React Component
+
+To use the SVG as an `svg` element and not a source for an `image` element, we will import the SVG as a ReactComponent. You can see an example of importing the SVG as a ReactComponent [here](https://github.com/layer5io/layer5/blob/master/src/sections/General/Navigation/index.js)
+
+```
+import { ReactComponent as Logo } from "../../../assets/images/app/layer5-colorMode.svg";
+```
+
+Then we include the component where you want to display it
+
+```
+<Link to="/" className="logo">
+    <Logo />
+</Link>
+```
+
+#### Adding the color change property to the SVG
+
+To change a fill for a SVG depending on color mode, you will need to create a class in the SVG to fill inner paths or rect or polygon with default colors, then assign that SVG class in a styled-component to a property value as done in components.
+
+Example #1 Layer5 Logo
+
+To see the code for this SVG click [here](https://github.com/layer5io/layer5/blob/master/src/assets/images/app/layer5-colorMode.svg?short_path=e76da80). (note if link is not working , click [here](https://github.com/layer5io/layer5/blob/master/src/assets/images/app/layer5-colorMode.svg) and then click the "<>" button to display the source blob.)
+
+In the SVG code, we see classes declared here:
+
+```
+<defs>
+        <style>
+            .colorMode1 {
+                fill: #fff
+            }
+          ...
+        </style>
+</defs>
+```
+
+We use `colorMode[#]` in the class name to designate classes that we want to change depending on color mode.
+
+Then in the related styled component, we use a specific class generated by styled-components to designate the theme property.
+
+In our example with the logo in the navigation, this is the relevant [styled component](https://github.com/layer5io/layer5/blob/master/src/sections/General/Navigation/navigation.style.js).
+
+And if we find target the SVG element in the appropriate class, for this example `.logo` is the relevant class. We then add the class that is generated by the styled-component plugin, which is based off the file name and the colorMode class name ( in this case `.layer5-colorMode_svg__colorMode1`).
```

Comment:
Here, if we want to update the CSS of the SVG then we target the appropriate parent class, ....

Reply:
Explain the file name.
Something like, (in this case `.layer5-colorMode_svg__colorMode1` where `layer5-colormode` is the file name, `_svg__` is the separator and `colorMode1` is the classname being targetted inside the SVG.